client/dashboard/sites/settings/site-actions.tsx

```diff
@@ -0,0 +1,101 @@
+import { useMutation } from '@tanstack/react-query';
+import { __experimentalHeading as Heading, Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { addQueryArgs } from '@wordpress/url';
+import { restoreSitePlanSoftwareMutation } from '../../app/queries';
+import { ActionList } from '../../components/action-list';
+import { DotcomFeatures } from '../../data/constants';
+import type { Site } from '../../data/types';
+
+const RestorePlanSoftware = ( { site }: { site: Site } ) => {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { slug, is_wpcom_atomic } = site;
+	const mutation = useMutation( restoreSitePlanSoftwareMutation( slug ) );
+
+	const handleClick = () => {
+		mutation.mutate( undefined, {
+			onSuccess: () => {
+				createSuccessNotice(
+					__( 'Requested restoration of plugins and themes that come with your plan.' ),
+					{ type: 'snackbar' }
+				);
+			},
+			onError: () => {
+				createErrorNotice( __( 'Failed to request restoration of plan plugin and themes.' ), {
+					type: 'snackbar',
+				} );
+			},
+		} );
+	};
+
+	if ( ! is_wpcom_atomic ) {
+		return null;
+	}
+
+	return (
+		<ActionList.ActionItem
+			title={ __( 'Re-install plugins & themes' ) }
+			description={ __(
+				'If your website is missing plugins and themes that come with your plan you can re-install them here.'
+			) }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					isBusy={ mutation.isPending }
+					onClick={ handleClick }
+				>
+					{ __( 'Restore' ) }
+				</Button>
+			}
+		/>
+	);
+};
+
+const DuplicateSite = ( { site }: { site: Site } ) => {
+	const { capabilities, plan, slug } = site;
+
+	if (
+		! ( capabilities.manage_options && plan?.features.active.includes( DotcomFeatures.COPY_SITE ) )
+	) {
+		return null;
+	}
+
+	return (
+		<ActionList.ActionItem
+			title={ __( 'Duplicate site' ) }
+			description={ __( 'Create a duplicate of this site.' ) }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					href={ addQueryArgs( '/setup/copy-site', {
+						sourceSlug: slug,
+					} ) }
+				>
+					{ __( 'Duplicate' ) }
+				</Button>
+			}
+		/>
+	);
+};
+
+export default function SiteActions( { site }: { site: Site } ) {
+	const actions = [
+		<RestorePlanSoftware key="restore-plan-software" site={ site } />,
+		<DuplicateSite key="duplicate-site" site={ site } />,
+	].filter( Boolean );
+
+	if ( ! actions.length ) {
```

Comment:
At the moment this check is useless. The filter function above as well. Elements are always defined. They're not evaluated here.

Reply:
To avoid complex code, potentially you could use `:has` CSS selector to hide the wrapper "div" if there are not visible actions or things like that.

Alternatively, don't use sub components and just inline all the actions here, which would allow you to use the "boolean" checks.

Reply:
>  Elements are always defined. They're not evaluated here.

Oops... I totally forgot that 🤦‍♂️ Will update it!